Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ILLD7YIGPositionCalibration.py

```diff
@@ -0,0 +1,517 @@
+
+        if self.getProperty('DetectorFitOutput').isDefault:
+            pass #will clean up the remaining workspaces
+        else:
+            self.setProperty('DetectorFitOutput', mtd['det_out_Parameters'])
+
+    def _get_scan_data(self):
+        """ Loads YIG scan data, removes monitors, and prepares
+        a workspace for Bragg peak fitting"""
+
+        # workspace indices for monitors
+        MONITOR_INDICES = "{0}, {1}".format(self._D7NumberPixels, self._D7NumberPixels+1)
+
+        outGroup = Load(self.getPropertyValue("Filenames"),
+                        OutputWorkspace="outGroup",
+                        PositionCalibration='None')
+        # load the group into a single table workspace
+        nfiles = outGroup.getNumberOfEntries()
+        # new vertical axis
+        x_axis = NumericAxis.create(nfiles)
+        # Fill the intensity and position tables with all the data from scans
+        for scan in range(nfiles):
+            workspace = outGroup.getItem(scan)
+            # normalize to monitor1 as monitor2 is sometimes empty:
+            if workspace.readY(workspace.getNumberHistograms()-2)[0] != 0:
+                workspace /= workspace.readY(workspace.getNumberHistograms()-2)[0]
+            # remove Monitors:
+            workspace = RemoveSpectra(workspace, MONITOR_INDICES)
+            # prepare proper label for the axes
+            x_axis_label = workspace.run().getProperty('2theta.requested').value
+            x_axis.setValue(scan, x_axis_label)
+            # convert the x-axis to signedTwoTheta
+            workspace = ConvertAxisByFormula(workspace,
+                                             Axis='X',
+                                             Formula='-180.0 * signedtwotheta / pi')
+            # mask detectors too close to the beam axis or too far away:
+            workspace = MaskBinsIf(workspace, Criterion='x>{0} && x<{1}'.format(self._beamMask1, self._beamMask2))
+            # append the new row to a new MatrixWorkspace
+            workspace = ConvertToPointData(InputWorkspace=workspace)
+            try:
+                intensityWS
+            except NameError:
+                intensityWS = workspace.clone()
+            else:
+                list = [intensityWS, workspace]
+                intensityWS = ConjoinXRuns(list)
+
+        #replace axis and corrects labels
+        x_axis.setUnit("Label").setLabel('TwoTheta', 'degrees')
+        intensityWS.replaceAxis(0, x_axis)
+        y_axis = NumericAxis.create(self._D7NumberPixels)
+        for pixel_no in range(self._D7NumberPixels):
+            y_axis.setValue(pixel_no, pixel_no+1)
+        intensityWS.replaceAxis(1, y_axis)
+        return intensityWS
+
+    def _load_yig_peaks(self, ws):
+        """Loads YIG peaks provided as an XML Instrument Parameter File"""
+        parameterFilename = self.getPropertyValue('YIGPeaksFile')
+        ClearInstrumentParameters(ws) #in case other IPF was loaded there before
```

Comment:
This may be an issue in case user provides an `InputWorkspace` that already had an IPF with YIG d-spacing loaded in. Then, loading a different IPF that contains slightly different d-spacings may cause a problem. An alternative would be to resign from providing YIG d-spacing as an IPF but read it directly as an XML file.